### DISCUSSION TITLE
Is Fyne worth our time?  That's a rhyme and it should be!

### DISCUSSION THREAD

**Discussion by kmesiab** [kmesiab/ai-code-critic#56] [General]
I'm finding the Fyne ui toolkit limiting.  I've looked into [wails.io](https://wails.io/docs/howdoesitwork) and was able to mock up a decent looking interface.

Seeing as the ui can be built using a react app, we can gain a great deal of styling flexibility.  

I will post a sample app later.

**Comment by kmesiab** [comment]
As of 1/14/2024, do not upgrade your fyne package to `v2.4.3` as the font package is broken:

`../../../../pkg/mod/fyne.io/fyne/v2@v2.4.3/internal/painter/font.go:209:65: out.LineBounds.LineHeight undefined (type shaping.Bounds has no field or method LineHeight)`

This is the second broken release and we may not be able to count on the reliability of this UI framework.

**Reply by kmesiab** [reply]
Ironically this bug led me to discover we do indeed have the ability to alter the line height. lol

**Comment by frankie-mur** [comment]
I recently listened to a great podcast episode with the creator of wails.io, it looks promising. Interested in seeing the mockup! I believe the current UI is very simple...in a good way. What is your overall vision for the app where you see a use case for an expanded UI?

**Reply by kmesiab** [reply]
I agree, and that was/is the goal (Simplicity).  

The only thing that pushes me for a change from Fyne is the poor presentation of the Markdown.  I explored writing a custom styler for it, but decided I was trying to force this library to do something it is not ready to do yet. 

I would have considered sticking with Fyne if there was a reasonable web-view component where we could retain greater control over the style, but alas, the work I've seen on that front is in earnest and not in a position to be a viable choice, or mature enough to bother contributing yet.

I like Wails in that we can develop the front end in React, which immediately presents a much richer playground for the app in a mobile platform.

I admittedly have not written React, though I do have some professional experience in Angular.   I don't anticipate a challenging learning curve.  

I ditched the wails prototpye because it was hacked together, but porting it to the new framework was negligible.  As an example, you can see how I've ported it to this console app:  https://github.com/kmesiab/cadre.  More importantly, you can see the difference in the quality of the presentation of the code review.  This is how the reports SHOULD look rendered by default.   The only reason I haven't continued to develop the wails prototype is time.  

At the end of the day, the app consists of two libraries: 

1.) The GitHub parser and 
2.) The tiny wrapper around the OpenAI library. 

If you wanted to try it out (a little wails ui)... what I did was simply modify the button click on the hello world app to call our GitHub library and output the contents wholesale.  

I was able to rig it up in a few minutes, which is a positive sign. lol.

Are you interested in exploring the wails.io direction, or do you like the idea of trying to optimize the Fyne RichText styles?  

I am not opposed to either direction, as long as the end result is a beautifully presented, easy to read, actionable code review.

It's also not the only option.  Since it is negligible to launch an extremely lightweight and performant local web server w/ go, it is not unthinkable to write the app as a self contained rest service serving a modern web front end wrapped in a desktop window.  

Or some similar strategy.

**Reply by frankie-mur** [reply]
Fyne is a little tricky, I did bang my hang a little over understanding the initial implementation, mostly dealing with Fyne and their docs. I like your idea...separating the two will also allow a much less steep learning curve, therefore promoting more collaborators. I think this is a win-win. I also don't have much React exp but it also supports many other frontend frameworks (Svelte, React, Vue, Preact, Lit, and Vanilla JS).

**Reply by kmesiab** [reply]
There's no reason we can't launch a parallel effort that we anticipate will supplant this one.  

We should preserve this repo and support those who've forked it.  

You pick a front end framework you feel most confident and I'll learn it.

Keep an eye on portability to mobile in the future! 

You in?

**Reply by frankie-mur** [reply]
Heck ya!

**Reply by kmesiab** [reply]
Great.  Here comes the hardest part.  Let's name the thing and I'll fire up a new repo with our necessary linters and GitHub actions.  Get our branches in order and I'll make you an admin. 

Brainstorm it up and drop some suggestions here.  I'm not too concerned about it, as long as it sounds cool and we can make a fun mascot for it 😉

**Reply by frankie-mur** [reply]
You mean you don't like `ai-code-critic` ?!
Jokes aside - yes! My favorite part of any app,  I will think of some and drop them here shortly, would like to hear some of your ideas as well! ⚡

**Reply by kmesiab** [reply]
How about:

CodeSight (a play on code and insight)
CodeInsightAI
AICodeGuru
InsightForge
CodeClarityAI
CodeIntelAI
CodeIntel
VisionCodeAI

Some variation some something like this perhaps?

**Reply by frankie-mur** [reply]
- Enki
- Orion
- Clio
- Quanta
- Syntax
- Parsec
- Semantix
- GitGonic
- PullPulse

**Reply by kmesiab** [reply]
I definitely like Orion. However, Syntax sounds cooler and doesn't actually have much usage out there after a quick Google search.

I'll go with any of:

Orion
Syntax
Semantix

**Reply by frankie-mur** [reply]
I like Orion!

**Reply by kmesiab** [reply]
https://github.com/kmesiab/orion

Go add yourself to the README
